[Merged by Bors] - Create a local testnet (#2614)
sigp/lighthouse: scripts/local_testnet/README.md

## Starting the testnet

Modify `vars.env` as desired.

> Start a local eth1 ganache server plus boot node along with NODE_COUNT

Reviewer: ```suggestion
Start a local eth1 ganache server plus boot node along with `NODE_COUNT`
```

Author: Done, but changed to `BN_COUNT`

> number of beacon nodes and VC_COUNT validator clients.

Reviewer: ```suggestion
number of beacon nodes and `VC_COUNT` validator clients.
```

Author: Done

> This script takes two options `-c VC_COUNT` and `-d DEBUG_LEVEL`.

Reviewer: ```suggestion
The `start_local_testnet.sh` script takes two options `-c VC_COUNT` and `-d DEBUG_LEVEL`.
```

Author: Done, but changed `-c` to `-v`

The options maybe in any order or absent in which case they take the default value specified.

> - VC_COUNT: the number of validator clients to create, default: `1`

Reviewer: I think the default here should be `VC_COUNT == NODE_COUNT` as we are distributing the validators among all nodes equally. 
It might also be easier to have this param in `vars.env`.
Also, having just 1/4th of the validators active would prevent finality, so we wouldn't want that to be default.

Author: Done, I've add VC_COUNT to vars.env and default it to BN_COUNT.

## Starting the testnet

Modify `vars.env` as desired.

Start a local eth1 ganache server plus boot node along with `BN_COUNT`
number of beacon nodes and `VC_COUNT` validator clients.

The `start_local_testnet.sh` script takes three options `-v VC_COUNT`, `-d DEBUG_LEVEL` and `-h` for help.

> The options maybe in any order or absent in which case they take the default value specified.

Reviewer 2: ```suggestion
The options may be in any order or absent in which case they take the default value specified.
```

Author: Done

- VC_COUNT: the number of validator clients to create, default: `BN_COUNT`
- DEBUG_LEVEL: one of { error, warn, info, debug, trace }, default: `info`

```bash
./start_local_testnet.sh
```

> ## Stoping the testnet

Reviewer 2: ```suggestion
## Stopping the testnet
```

Author: Done

This is not necessary before `start_local_testnet.sh` as it invokes `stop_local_testnet.sh` automatically.
```bash
./stop_local_testnet.sh
```

## Manual creation of local testnet

> These scripts used by ./start_local_testnet.sh and maybe used to manually

Reviewer 2: ```suggestion
These scripts are used by ./start_local_testnet.sh and may be used manually
```

Author: Done
